ex5/dist/main.js

```diff
@@ -0,0 +1,156 @@
+class Main {
+	constructor() {
+		this.itemClient = new ItemClient();
+	}
+
+	init = async () => {
+		const addItemButton = document.getElementById('list-item-submit');
+		addItemButton.addEventListener('click', this.handleItem);
+
+		const deleteAllBtn = document.getElementById('list-item-clear');
+		deleteAllBtn.addEventListener('click', this.handleDeleteAll);
+
+		await this.renderItems();
+	};
+
+	handleItem = async () => {
+		const input = document.getElementById('list-item-input');
+		const inputValue = input.value;
+		await this.itemClient.postItem(inputValue);
+		await this.renderItems();
+	};
+
+	deleteItem = async (item) => {
+		await this.itemClient.deleteItem(item);
+		await this.renderItems();
+	};
+	popUp = async (item) => {
+		const { value: text } = await swal.fire({
+			input: 'textarea',
+			inputLabel: 'Edit your task',
+			inputValue: item,
+			inputAttributes: {
+				'aria-label': 'Type your message here',
+			},
+			showCancelButton: true,
+		});
+		if (text) {
+			await this.itemClient.editItem(item, text);
+		}
+	};
+	editItem = async (item) => {
+		let itemToEdit;
+		const list = document.querySelectorAll('.list-item');
+		for (const listItem of list) {
+			if (listItem.textContent == item) {
+				itemToEdit = listItem;
```

Comment:
you're not actually using the itemToEdit, since you have the item in hand you could just call this.popUp(item); without the loop